packages/vc-verification/src/resolver/idata-cache.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * An Interface for Data aggreator to cache responses from Blockchain and SSI-Hub
+ */
+export interface IDataCache {
+  /**
+   * Stores role credential
+   * @param did user DID
+   * @param role namespace
+   * @param data Credential of the user for the namespace
+   */
+  setRoleCredential(did: string, role: string, data: any): void;
+
+  /**
+   * Stores DID Document
+   * @param did user DID
+   * @param data DIDDocument of the user (DID)
+   */
+  setDIDDocument(did: string, data: any): void;
+
+  /**
+   * Stores RoleDefinition
+   * @param role namespace for which definition is being stored
+   * @param data role definition
+   */
+  setRoleDefinition(role: string, data: any): void;
+
+  /**
+   * Returns cached role credential
+   * @param did user DID
+   * @param role namespace
+   */
+  getRoleCredential(did: string, role: string): any | undefined;
+
+  /**
+   * Returns cached DID Document (services)
+   * @param did user DID
+   */
+  getDIDDocument(did: string): any | undefined;
+
+  /**
+   * Returns cached role defintion
+   * @param role namespace
+   */
+  getRoleDefinition(role: string): any | undefined;
+
+  /**
+   * Clears all cached data
+   */
+  clearData(): void;
```

Comment:
Also, I see this `clearData` function isn't being used anywhere. IMO, from a YAGNI principle perspective, it should be removed.